### PR TITLE
ruff: 0.0.259 -> 0.0.260

### LINKED ISSUE #223252: `ruff` causes Illegal instruction: 4 on `x86_64-darwin`
### Describe the bug

[ruff](https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/tools/ruff/default.nix) `0.0.259` causes  **Illegal instruction: 4** on x86 darwin.

### Steps To Reproduce

On a few recent nixos-unstable commits, e.g.:

```console
$ nix shell --no-use-registries github:NixOS/nixpkgs/8f40f2f90b9c9032d1b824442cfbbe0dbabd0dbd#ruff -c ruff
Illegal instruction: 4

$ nix shell github:NixOS/nixpkgs/8f40f2f90b9c9032d1b824442cfbbe0dbabd0dbd#ruff -c which ruff
/nix/store/vrxifpk5bhdgrq28qn9yna63c4w62v08-ruff-0.0.259/bin/ruff
```

### Expected behavior

`ruff` outputs command line options.

### Additional context

I wonder how it interacts with `rustPlatform` or it's a problem in the package. Impurity was a reason for similar problems (https://github.com/NixOS/nixpkgs/issues/58950). 

I rebuilt `ruff` in gh runners, which worked fine: https://github.com/ahirner/nix-trickle/commit/5610568dfae8633ef6228df3619bffd9a12b000c
A local cargo release build of `ruff` works. 
An older version from the binary cache works as well:
```console
$ nix shell --no-use-registries github:NixOS/nixpkgs/a1291d0d020a200c7ce3c48e96090bfa4890a475#ruff -c ruff --version
ruff 0.0.244
```

### Notify maintainers

@figsoda 

### Metadata

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 22.2.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.14.1`
 - channels(root): `"nixpkgs"`
 - channels(user): `""`
 - nixpkgs: `/etc/nix/inputs/nixpkgs`
```

I'm running darwin-nixos and the system is somehow reported wrongly.
system_info

```
$ sysctl -a | grep 'machdep.cpu.*features'
machdep.cpu.features: FPU VME DE PSE TSC MSR PAE MCE CX8 APIC SEP MTRR PGE MCA CMOV PAT PSE36 CLFSH DS ACPI MMX FXSR SSE SSE2 SS HTT TM PBE SSE3 PCLMULQDQ DTES64 MON DSCPL VMX EST TM2 SSSE3 FMA CX16 TPR PDCM SSE4.1 SSE4.2 x2APIC MOVBE POPCNT AES PCID XSAVE OSXSAVE SEGLIM64 TSCTMR AVX1.0 RDRAND F16C
machdep.cpu.leaf7_features: RDWRFSGS TSC_THREAD_OFFSET SGX BMI1 AVX2 FDPEO SMEP BMI2 ERMS INVPCID FPU_CSDS AVX512F AVX512DQ RDSEED ADX SMAP AVX512IFMA CLFSOPT IPT AVX512CD SHA AVX512BW AVX512VL AVX512VBMI UMIP PKU GFNI VAES VPCLMULQDQ AVX512VNNI AVX512BITALG AVX512VPOPCNTDQ RDPID SGXLC FSREPMOV MDCLEAR IBRS STIBP L1DF ACAPMSR SSBD
machdep.cpu.extfeatures: SYSCALL XD 1GBPAGE EM64T LAHF LZCNT PREFETCHW RDTSCP TSCI
```

### PR DESCRIPTION
**Pull request by figsoda**
Diff: https://github.com/charliermarsh/ruff/compare/v0.0.259...v0.0.260

Changelog: https://github.com/charliermarsh/ruff/releases/tag/v0.0.260

###### Description of changes

closes #223252

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/ruff/default.nix [hunk 1]
```diff
@@ -38,9 +38,7 @@ rustPlatform.buildRustPackage rec {
   ];
 
   cargoBuildFlags = [ "--package=ruff_cli" ];
-
-  # building tests fails with `undefined symbols`
-  doCheck = false;
+  cargoTestFLags = cargoBuildFlags;
```
Comment (zowoq) [inline pkgs/development/tools/ruff/default.nix:41]:
```suggestion
  cargoTestFlags = cargoBuildFlags;
```

**Comment by zowoq** [comment]
Would using the newer sdk fix the x86_64-darwin failure?

```diff
diff --git a/pkgs/development/tools/ruff/default.nix b/pkgs/development/tools/ruff/default.nix
index 38e6a3722b7..a25217ad1ed 100644
--- a/pkgs/development/tools/ruff/default.nix
+++ b/pkgs/development/tools/ruff/default.nix
@@ -34,7 +34,7 @@ rustPlatform.buildRustPackage rec {
   ];
 
   buildInputs = lib.optionals stdenv.isDarwin [
-    darwin.apple_sdk.frameworks.CoreServices
+    darwin.apple_sdk_11_0.frameworks.CoreServices
   ];
 
   cargoBuildFlags = [ "--package=ruff_cli" ];
diff --git a/pkgs/top-level/all-packages.nix b/pkgs/top-level/all-packages.nix
index 41ffa819427..dc218cb13a3 100644
--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -38605,7 +38605,7 @@ with pkgs;
 
   rucksack = callPackage ../development/tools/rucksack { };
 
-  ruff = callPackage ../development/tools/ruff { };
+  ruff = darwin.apple_sdk_11_0.callPackage ../development/tools/ruff { };
 
   sam-ba = callPackage ../tools/misc/sam-ba { };
 

```

**Comment by figsoda** [comment]
> Would using the newer sdk fix the x86_64-darwin failure?

Doesn't seem to work. The issue was already present (https://github.com/NixOS/nixpkgs/issues/223252), we are just getting the errors now because tests were disabled. @ahirner had success building it on github actions, so it is probably better to keep the tests enabled so failure happens build time and hydra doesn't cache something bad. Not sure what impurity got in that might have caused this though

**Comment by zowoq** [comment]
> Doesn't seem to work.

Which did you test, native or emulated on aarch64-darwin?

**Comment by figsoda** [comment]
emulated on aarch64-darwin (darwin-build-box)

**Comment by figsoda** [comment]
now ofborg is passing, maybe it only fails when its emulated?

**Comment by zowoq** [comment]
Looks like it's `jemalloc` again. https://github.com/NixOS/nixpkgs/pull/193403

```diff
diff --git a/pkgs/development/tools/ruff/default.nix b/pkgs/development/tools/ruff/default.nix
index e732e6f24c8..7454671b4ab 100644
--- a/pkgs/development/tools/ruff/default.nix
+++ b/pkgs/development/tools/ruff/default.nix
@@ -37,6 +37,12 @@ rustPlatform.buildRustPackage rec {
     darwin.apple_sdk.frameworks.CoreServices
   ];
 
+  preBuild = lib.optionalString (stdenv.isDarwin && stdenv.isx86_64) ''
+    # See https://github.com/jemalloc/jemalloc/issues/1997
+    # Using a value of 48 should work on both emulated and native x86_64-darwin.
+    export JEMALLOC_SYS_WITH_LG_VADDR=48
+  '';
+
   cargoBuildFlags = [ "--package=ruff_cli" ];
   cargoTestFlags = cargoBuildFlags;
 

```

**Comment by figsoda** [comment]
@zowoq thanks for debugging, that seems to fix it. I would've never though of that

**Comment by zowoq** [comment]
This is probably an issue for every rust package if it vendors jemalloc and it gets built via emulation.

I'll see if I can find any, if there are others I guess we'll want to put the fix in `buildRustPackage`.

**Pull request merged**